src/settings_gui.cpp

```diff
@@ -547,7 +550,8 @@ struct GameOptionsWindow : Window {
 
 			case WID_GO_FONT_ZOOM_DROPDOWN:
 				GfxClearSpriteCache();
-				_font_zoom = (ZoomLevel)(ZOOM_LVL_OUT_4X - index);
+				_font_zoom_cfg = index > 0 ? ZOOM_LVL_OUT_4X - index + 1 : ZOOM_LVL_GUI_AUTO;
```

Comment:
In general I like this PR, but the implementation feels a bit awkward, and I have been wondering why I feel that way. And I think it comes down to the way that "gui" variables and non-"gui" variables are now mixed with signed vs unsigned variables and with `< ZOOM_LVL_MIN` with `ZOOM_LVL_GUI_AUTO`.

I think you went this road because of the settings system, where you need a `min` and `max` for every setting, and didn't want to use a value like `4` for auto-detect. Which I can understand. But the code now becomes a bit difficult to follow, more so than it already does (the zoom-code is really hard to follow, because of the way it was implemented). But bumping all current zoom-levels by one is also sub-optimal, of course.

Two things that stand out: `ZOOM_LVL_GUI_AUTO` seems like a weird name to pick, as it conflicts with `gui_zoom` vs `_font_zoom`. Suggestion: `ZOOM_LVL_AUTO` or `ZOOM_LVL_CFG_AUTO`.

Next, is a bit above, where you do `if (_gui_zoom_cfg < ZOOM_LVL_MIN) {`, which implies that `ZOOM_LVL_GUI_AUTO` should be a lower value than `ZOOM_LVL_MIN`, but as I mentioned in another comment, is not enforced in the code itself. Suggestion: `if (_gui_zoom_cfg == ZOOM_LVL_GUI_AUTO) {`.

I think that these changes alone are enough to make it a bit more clear what is going on.